storybookjs/storybook
# Preact: Add react compat by default

Issue:  #12853

implements #12403 and adds React component demos to the Preact kitchen sink

## What I did

- ported changes from #12403 into a fresh branch
- added a Storybook story to the Preact kitchen sink to illustrate that React components work.
- fixed preact storyshots to work with hooks when using `React` components

## How to test

- Is this testable with Jest or Chromatic screenshots? ✔ I added a React example that renders both a React functional and class component in a story.
- Does this need a new example in the kitchen sink apps? ✔ As stated above.
- Does this need an update to the documentation? ❌ Maybe a mention that React compat is enabled by default?

If your answer is yes to any of these, please make sure to include it in your PR.
